tests/subset.rs

```diff
@@ -45,21 +45,19 @@ fn test_subset<A: Adversary<Subset<NodeId>>>(
     // Verify that all instances output the same set.
     let mut expected = None;
     for node in network.nodes.values() {
-        if let Some(output) = expected.as_ref() {
-            assert!(once(output).eq(node.outputs()));
-            continue;
-        }
-        assert_eq!(1, node.outputs().len());
-        expected = Some(node.outputs()[0].clone());
+        assert!(1 < node.outputs().len());
+        expected = Some(node.outputs().clone());
     }
     let output = expected.unwrap();
-    assert!(once(&output).eq(network.observer.outputs()));
+
     // The Subset algorithm guarantees that more than two thirds of the proposed elements
     // are in the set.
-    assert!(output.len() * 3 > inputs.len() * 2);
-    // Verify that the set's elements match the proposed values.
-    for (id, value) in output {
-        assert_eq!(inputs[&id], value);
+    assert!((output.len() - 1) * 3 > inputs.len() * 2);
+    for i in output {
```

Comment:
I'd remove this loop now. Instead, we should check in the loop above _for each node_ that the `Contribution` values are the same set, and that `Done` is output exactly once and at the end.

Reply:
Done